test/e2e/tests/utils/utils.go

```diff
@@ -47,32 +47,45 @@ func VerifyCheck(c *assert.CollectT, collectorOutput string, checkName string) {
 
 	checksJson := common.ParseCollectorJson(collectorOutput)
 	if checksJson != nil {
-		runnerStats, ok := checksJson["runnerStats"].(map[string]interface{})
-		assert.True(c, ok)
-		assert.NotNil(c, runnerStats)
+		runnerStats, runnerStatsOk := checksJson["runnerStats"].(map[string]interface{})
+		if !runnerStatsOk {
+			assert.Fail(c, "runnerStats field is not a map or is nil")
+			return
+		}
 
-		runningChecks, ok = runnerStats["Checks"].(map[string]interface{})
-		assert.True(c, ok)
-		assert.NotNil(c, runningChecks)
+		var checksOk bool
+		runningChecks, checksOk = runnerStats["Checks"].(map[string]interface{})
+		if !checksOk {
+			assert.Fail(c, "Checks field is not a map or is nil")
+			return
+		}
 
 		if check, found := runningChecks[checkName].(map[string]interface{}); found {
 			for _, instance := range check {
-				assert.Equal(c, checkName, instance.(map[string]interface{})["CheckName"].(string))
+				instanceMap, instanceOk := instance.(map[string]interface{})
+				if !instanceOk {
+					continue
+				}
 
-				lastError, exists := instance.(map[string]interface{})["LastError"].(string)
+				checkNameVal, checkNameOk := instanceMap["CheckName"].(string)
+				if checkNameOk {
+					assert.Equal(c, checkName, checkNameVal)
+				}
+
+				lastError, exists := instanceMap["LastError"].(string)
 				assert.True(c, exists)
 				assert.Empty(c, lastError)
 
-				totalErrors, exists := instance.(map[string]interface{})["TotalErrors"].(float64)
+				totalErrors, exists := instanceMap["TotalErrors"].(float64)
 				assert.True(c, exists)
 				assert.Zero(c, totalErrors)
 
-				totalMetricSamples, exists := instance.(map[string]interface{})["TotalMetricSamples"].(float64)
+				totalMetricSamples, exists := instanceMap["TotalMetricSamples"].(float64)
 				assert.True(c, exists)
 				assert.Greater(c, totalMetricSamples, float64(0))
 			}
 		} else {
-			assert.True(c, found, "Check %s not found or not yet running.", checkName)
+			assert.Failf(c, "Check not found", "Check %s not found or not yet running", checkName)
```

Comment:
In the `else` branch here, `found` is false, so we'd guarantee the assertion fails here. Let's explicitly fail for clarity